### ISSUE TITLE
[Tabs][a11y] Disclosure navigation not accessible via keyboard navigation (tab key)

### ISSUE THREAD

**Issue by ginabak** [Shopify/polaris-react#3673]
## Issue summary

While navigating tabs via keyboard, the disclosure tab is not accessible (while `tabbing`) 

## Expected behavior

This is what it should be ["acting" like](https://www.w3.org/TR/wai-aria-practices-1.1/examples/disclosure/disclosure-navigation.html). 

The user should be able to access the disclosure tab using the `tab` key and press enter to select what is rendered when selected. 

## Actual behavior
<details>
<summary>Related GIF</summary>

See the gif below - _on Zoom on MacOS (FireFox_):
![ZoomTabs](https://user-images.githubusercontent.com/43223543/100694455-26270a80-335d-11eb-996b-e821bf42cdcd.gif)

**Description**: I'm using arrow keys to navigate from tab to tab but when I get to the last one (before the disclosure), it goes back to the first tab ... 🤭, I can't `tab` into the disclosure tab
</details>

## Steps to reproduce the problem

1. Go to http://polaris-react.herokuapp.com/?path=/story/all-components-tabs--all-examples
2. Enable `New Design Language`
3. Make sure the canvas is small enough so that the disclosure tab is rendered. 
4. Navigate the tabs using the keyboard
5. To see the issue, press `tab`, one should be able to `tab` into the disclosure

## Reduced test case
 
1. Go to https://codesandbox.io/s/magical-goldstine-s1hki?file=/index.js 
2. Ensure that ``` <AppProvider i18n={en} features={{ newDesignLanguage: true }}>```
3. Make your window small enough so that you see a disclosure being rendered, like this: 
<img width="345" alt="screenshot of tabs with disclosure tab" src="https://user-images.githubusercontent.com/43223543/100791534-5f0bc180-33e7-11eb-9c7f-45c8bddfb27c.png">
4. Start navigating tabs using keyboard (tab to get focused, then use arrow keys to navigate back and forth), technically, the disclosure tab should be `tabbable` only. 

## Specifications

- Are you using the React components? (Y/N): Y
- Polaris version number: 5.10.1
- Browser: Chrome & FireFox
- Device: Mac
- Operating System: 10.15.7
